waspc/cli/Command/Watch.hs

```diff
@@ -0,0 +1,73 @@
+module Command.Watch
+    ( watch
+    ) where
+
+import           Control.Concurrent.Chan (Chan, newChan, readChan)
+import           Data.List               (isSuffixOf)
+import           Data.Time.Clock         (UTCTime, getCurrentTime)
+import qualified System.FilePath         as FP
+import qualified System.FSNotify         as FSN
+
+import           Command.Compile         (compile)
+import           Common                  (waspSays)
+import qualified Common
+import qualified Lib
+import           StrongPath              (Abs, Dir, Path, (</>))
+import qualified StrongPath              as SP
+
+
+-- TODO: Another possible problem: on re-generation, wasp re-generates a lot of files, even those that should not
+--   be generated again, since it is not smart enough yet to know which files do not need to be regenerated.
+--   This can trigger `npm start` processes to reload multiple times, once for each file!
```

Comment:
haven't we solved this at the end with a channel from which you only take the first event and disregard the others?

Reply:
No, with channels we solved the problem where you edit B.wasp and C.wasp while A.wasp was already being recompiled, and then it recompiles two times extra, once for B and once for C, instead of doing it just once for B and C at the same time.

Here we have problem of you editing A.wasp and Wasp generates whole new server/, which is multiple files, and then `nodemon` says "oh many files changed at once, let me restart once for each one". But ok, `nodemon` is smarter than that luckily, so it is ok for now. It might become bigger problem once Wasp apps grow bigger.